cni/pkg/nodeagent/informers_test.go

```diff
@@ -384,6 +384,16 @@ func TestExistingPodRemovedWhenPodAnnotated(t *testing.T) {
 
 	assertPodNotAnnotated(t, client, pod)
 
+	// path a test label to emulate a non-annotation POD update event
+	_, err = client.Kube().CoreV1().Pods(pod.Namespace).Patch(ctx, pod.Name,
+		types.MergePatchType, []byte(`{"metadata":{"labels":{"test":"update"}}}`), metav1.PatchOptions{})
+	assert.NoError(t, err)
+
+	// wait for an update events
+	mt.Assert(EventTotals.Name(), map[string]string{"type": "update"}, monitortest.AtLeast(5))
```

Comment:
Aha, now I see. Good catch. This should be fine in practice as the removal will be a no-op but either way, it is good to have a test for this.